### PR TITLE
BH-51802 call hideResults when we remove the search symbol

### PR DESCRIPTION
**Pull request by nathan-wilson-bh**
##### **Reviewers**
* @jgodi
* @more

### CODE REVIEW

File: src/elements/quick-note/QuickNote.ts [hunk 1]
```diff
@@ -394,6 +394,11 @@ export class QuickNoteElement extends OutsideClick implements OnInit, OnDestroy,
         let word = this.getWordAtCursor().trim();
         if (this.isTagging) {
             let symbol = this.config.triggers[this.taggingMode];
+            if (word.indexOf(symbol) === -1) {
+                this.zone.run(() => {
```
Comment (ndickerson) [inline src/elements/quick-note/QuickNote.ts]: We don't need the zone.run here - at this point, we are already in the Angular context.

File: src/elements/quick-note/QuickNote.ts [hunk 2]
```diff
@@ -394,6 +394,11 @@ export class QuickNoteElement extends OutsideClick implements OnInit, OnDestroy,
         let word = this.getWordAtCursor().trim();
         if (this.isTagging) {
             let symbol = this.config.triggers[this.taggingMode];
+            if (word.indexOf(symbol) === -1) {
```
Comment (ndickerson) [inline src/elements/quick-note/QuickNote.ts]: Can we do a `!word.includes(...)` instead?

File: src/elements/quick-note/QuickNote.ts (hunk 1 above)
Comment (nathan-wilson-bh) [inline src/elements/quick-note/QuickNote.ts]: I took out the zone.run and called hideResults directly and the component never actually gets destroyed

Comment (ndickerson) [inline src/elements/quick-note/QuickNote.ts]: Ok. Can we add a unit test case for this?

Comment (nathan-wilson-bh) [inline src/elements/quick-note/QuickNote.ts]: Yeah I planned on doing that. Just wanted to get the implementation out there for review first

Comment (nathan-wilson-bh) [inline src/elements/quick-note/QuickNote.ts]: So turns out I didn't need to run it in the angular zone and it was somehow just masking a bigger issue. Now it just returns empty string after it hides and works without running it in the angular zone

File: src/elements/quick-note/QuickNote.ts (hunk 2 above)
Comment (nathan-wilson-bh) [inline src/elements/quick-note/QuickNote.ts]: Done.

File: src/elements/quick-note/QuickNote.ts [hunk 3]
```diff
@@ -394,6 +394,10 @@ export class QuickNoteElement extends OutsideClick implements OnInit, OnDestroy,
         let word = this.getWordAtCursor().trim();
         if (this.isTagging) {
             let symbol = this.config.triggers[this.taggingMode];
+            if (!word.includes(symbol)) {
```
Comment (jgodi) [inline src/elements/quick-note/QuickNote.ts]: Spec?

Comment (nathan-wilson-bh) [inline src/elements/quick-note/QuickNote.ts]: Coming in the next commit

Comment (nathan-wilson-bh) [inline src/elements/quick-note/QuickNote.ts]: Done

**Pull request merged**